### PR TITLE
HITL: Add HTTP availability server

### PR DESCRIPTION
**Pull request by eundersander**
## Motivation and Context

* Add HTTP availability server required by Amazon Elastic Load Balancer health checks.
* Refactor networking process main loop to use async/await more cleanly
* Add HTML stub client for easier testing

## How Has This Been Tested

Local testing and AWS testing with HTML stub client.

## Types of changes

- **\[HITL\]**

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/hitl/pick_throw_vr/config/experiment/headless_server.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 habitat_hitl:
   window: ~
+  networking:
```
Comment (eundersander) [inline examples/hitl/pick_throw_vr/config/experiment/headless_server.yaml:5]: See comments in hitl_defaults.yaml.

File: habitat-hitl/habitat_hitl/_internal/networking/networking_process.py [hunk 2]
```diff
@@ -25,38 +26,28 @@
...
     global networking_process
-    global exit_event
 
-    # multiprocessing.dummy.Process is actually a thread and requires special logic
-    # to terminate it.
-    exit_event = (
-        threading.Event() if multiprocessing_config.use_dummy else None
-    )
+    # we don't support dummy multiprocessing at this time
+    assert not multiprocessing_config.use_dummy
```
Comment (eundersander) [inline habitat-hitl/habitat_hitl/_internal/networking/networking_process.py:37]: It's not worth the complexity of maintaining the dummy-multiprocessing version of all this.

File: habitat-hitl/habitat_hitl/_internal/networking/networking_process.py [hunk 3]
```diff
@@ -257,29 +228,83 @@ async def serve(self, websocket):
...
-        asyncio.set_event_loop(loop)
+    return runner
 
-    network_mgr = NetworkManager(interprocess_record, exit_event)
-    network_mgr_lambda = lambda ws, path: network_mgr.serve(ws)
-    ssl_context = create_ssl_context() if use_ssl else None
-    start_server = websockets.serve(
-        network_mgr_lambda, "0.0.0.0", 8888, ssl=ssl_context
+
+async def networking_main_async(interprocess_record):
```
Comment (eundersander) [inline habitat-hitl/habitat_hitl/_internal/networking/networking_process.py:270]: The big refactor here is that all our networking logic is now implemented with `async/await`. Reviewers should take some time to make sure they understand how `networking_main_async` and `networking_main` work.

**Review by eundersander — COMMENTED** [review]
Some comments for reviewers

File: habitat-hitl/habitat_hitl/_internal/networking/networking_process.py (hunk 2 above)
Comment (0mdc) [reply habitat-hitl/habitat_hitl/_internal/networking/networking_process.py:37]: Should we just remove it, and introduce it again if we need it?

File: habitat-hitl/habitat_hitl/_internal/networking/networking_process.py [hunk 4]
```diff
@@ -257,29 +228,86 @@ async def serve(self, websocket):
...
-    asyncio.get_event_loop().run_until_complete(start_server)
-    print("NetworkManager started on server thread. Waiting for clients...")
-    while not (exit_event and exit_event.is_set()):
-        asyncio.get_event_loop().run_until_complete(
-            asyncio.sleep(1.0)
-        )  # todo: investigate what sleep duration does here
+    # Handle SIGTERM. We should get this signal when we do networking_process.terminate(). See terminate_networking_process.
+    stop: asyncio.Future = asyncio.Future()
+    loop = asyncio.get_event_loop()
+    loop.add_signal_handler(signal.SIGTERM, stop.set_result, None)
```
Comment (0mdc) [inline habitat-hitl/habitat_hitl/_internal/networking/networking_process.py:293]: 👍

**Review by 0mdc — APPROVED** [review]
LGTM!

I tested the networking server, and it works as expected.

**Pull request merged**